docs/user_guide/security/identity_security.rst

```diff
@@ -0,0 +1,503 @@
+
+    AFTER_CHECK_CLIENT_RESOURCES = "_after_check_client_resources"
+    DEPLOY_JOB_TO_SERVER = "_deploy_job_to_server"
+    DEPLOY_JOB_TO_CLIENT = "_deploy_job_to_client"
+
+    BEFORE_SEND_ADMIN_COMMAND = "_before_send_admin_command"
+    
+    BEFORE_CLIENT_REGISTER = "_before_client_register"
+    AFTER_CLIENT_REGISTER = "_after_client_register"
+    CLIENT_REGISTERED = "_client_registered"
+    SYSTEM_BOOTSTRAP = "_system_bootstrap"
+
+    AUTHORIZE_COMMAND_CHECK = "_authorize_command_check"
+
+
+Security check Inputs
+---------------------
+Make a ``SECURITY_ITEMS`` dict available in the FLContext, which holds any security check related data.
+
+NVFlare standard data:
+
+.. code-block:: python
+
+    IDENTITY_NAME
+    SITE_NAME
+    SITE_ORG
+    USER_NAME
+    USER_ORG
+    USER_ROLE
+    JOB_META
+
+
+Security check Outputs
+----------------------
+
+.. code-block:: python
+
+    AUTHORIZATION_RESULT
+    AUTHORIZATION_REASON
+
+
+NVFlare will check the AUTHORIZATION_RESULT to determine if the operations have been authorized to perform. Before each
+operation, NVFLare platform removes any AUTHORIZATION_RESULT in the FLContext. After the authorization check process, it
+looks for if these results are presented in the FLContext or not. If it presents, it uses its TRUE/FALSE value to determine the action. If not present, it will be treated as TRUE.
+
+Each FLComponent listening and handling the event can use the security data to generate the necessary authorization check
+results as needed. The workflow will only continue when all the FLComponents pass the security check. Any one FLComponent
+that has the FALSE value will cause the workflow to stop execution.
+
+
+Admin tool event support
+------------------------
+In order to support additional security data for site-specific customized authentication, we need to add the support for
+event based solutions for the admin console / API tool. Using these events, the admin tool will be able to add in the custom
+SSL certificates, etc, security related data, sent along with the admin commands to the server for site-specific authentication check.
+
+.. code-block:: python
+
+    BEFORE_ADMIN_REGISTER
+    AFTER_ADMIN_REGISTER
+    BEFORE_SENDING_COMMAND
+    AFTER_SENDING_COMMAND
+    BEFORE_RECEIVING_ADMIN_RESULT
+    AFTER_RECEIVING_ADMIN_RESULT
+
+The site-specific authentication and authorization applies to both admin too and Admin APIs.
+
+Allow more data to be sent to the server for client registration
+----------------------------------------------------------------
+If the application needs to send additional data from the client to the server to perform the authentication check, the client
+can set the data into the FL_Context as public data. Then the server side can get access to the data through the PEER_FL_CONTEXT.
+The application can build the FLComponent to listen to the EventType.CLIENT_REGISTERED to perform the authentication check needed.
+
+
+Site-specific Security Example
```

Comment:
maybe later we can move this to repo's example so that we can just put a link here to refer to it.

Reply:
can be a follow up PR @nvkevlu